c/src/concept/concept.rs

```diff
@@ -56,36 +56,59 @@ impl DatetimeInNanos {
     }
 }
 
-/// A <code>DatetimeAndZoneId</code> used to represent IANA time zoned datetime in FFI.
+/// A <code>DatetimeAndTimeZone</code> used to represent time zoned datetime in FFI.
+/// Time zone can be represented either as an IANA <code>Tz</code> or as a <code>FixedOffset</code>.
+/// Either the zone_name (is_fixed_offset == false) or offset (is_fixed_offset == true) is set.
 #[repr(C)]
-pub struct DatetimeAndZoneId {
+pub struct DatetimeAndTimeZone {
     datetime_in_nanos: DatetimeInNanos,
-    zone_id: *mut c_char,
-}
-
-impl DatetimeAndZoneId {
-    pub fn new(datetime_in_nanos: DatetimeInNanos, zone_id: String) -> Self {
-        Self { datetime_in_nanos, zone_id: release_string(zone_id) }
+    zone_name: *mut c_char,
+    local_minus_utc_offset: i32,
+    is_fixed_offset: bool,
```

Comment:
I could work with optional integers instead of this bool flag (`null` if not offset), but it's an additional memory and SWIG headache.